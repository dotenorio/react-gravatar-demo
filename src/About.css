a {
  color: #444;
}

a:hover {
  opacity: 0.6;
}