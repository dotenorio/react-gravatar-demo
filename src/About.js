import React from 'react'

import Layout from "./Layout";
import "./About.css";

function About () {
  return (
    <div>
      <Layout title="About This Demo">
        <p>A React Gravatar Demo..</p>
        <p>See more <a href="https://github.com/dotenorio/react-gravatar-demo" title="See this on Github">on Github</a>.</p>
      </Layout>
    </div>
  );
}

export default About;
