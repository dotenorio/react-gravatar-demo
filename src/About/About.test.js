import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import About from './About';

test('Load About', () => {
  const wrapper = shallow(
    <About />
  );

  expect(wrapper.contains(<p>A React Gravatar Demo..</p>)).toEqual(true);
  expect(wrapper.find('p').last().text()).toEqual('See more on Github.');
})

test('About renders correctly', () => {
  const tree = renderer
    .create(<MemoryRouter><About /></MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});