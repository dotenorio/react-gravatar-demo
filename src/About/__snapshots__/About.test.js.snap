// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly 1`] = `
<div>
  <div>
    <div
      className="header"
    >
      <h1>
        About This Demo
      </h1>
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
            title="Home Page"
          >
            Home
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/about"
            onClick={[Function]}
            title="About Project"
          >
            About
          </a>
        </li>
      </ul>
    </div>
    <div
      className="content"
    >
      <p>
        A React Gravatar Demo..
      </p>
      <p>
        See more 
        <a
          href="https://github.com/dotenorio/react-gravatar-demo"
          title="See this on Github"
        >
          on Github
        </a>
        .
      </p>
    </div>
  </div>
</div>
`;
